docs/docs/guides/build/components/building-pipelines-with-components/using-template-udfs.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Use template variables to inject user-defined functions into component definitions.
+sidebar_position: 350
+title: Using template UDFs 
+---
+
+Template variables can be used to create user-defined functions (UDFs). There is no formal concept of a UDF in dagster. It is just a convention to a describe template variable that is a function.
+
+This is a powerful feature that enables you to inject Python functions into your component YAML definitions, which are evaluated at definition load time, and the output of those functions are injected into the YAML document. This can occur anywhere in the document, allowing you to "mix and match" Python and YAML
+seamlessly.
+
+## Why UDFs instead of template logic?
+
+Many templating engines offer features like embedded conditionals (`{% if %}`) and loops (`{% for %}`) to handle these complex use cases. UDFs are an alternative to this. Instead of using `{% for %}` you can instead invoke a python function that has a `for` loop
```

Comment:
```suggestion
Many templating engines offer features like embedded conditionals (`{% if %}`) and loops (`{% for %}`) to handle complex use cases. UDFs are an alternative to this. Instead of using `{% for %}`, you can invoke a Python function that has a `for` loop.
```